compiler/qsc_data_structures/src/language_features.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use clap::ValueEnum;
+use serde::Deserialize;
+use std::collections::BTreeSet;
+#[derive(Deserialize, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Copy)]
+#[serde(rename_all = "kebab-case")]
+pub enum LanguageFeature {
+    /// This language feature enables experimental syntax that will likely be stabilized in the next major version.
+    /// It may include removals of outdated syntax and introductions of new syntax.
+    V2PreviewSyntax,
+}
+
+impl LanguageFeature {
+    pub fn try_parse(s: &str) -> Result<Self, UnrecognizedLanguageFeature> {
+        match s {
+            "v2-preview-syntax" => Ok(LanguageFeature::V2PreviewSyntax),
+            _ => Err(UnrecognizedLanguageFeature::msg(format!(
+                "Unrecognized language feature: {s}"
+            ))),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Default)]
+pub struct LanguageFeatures(BTreeSet<LanguageFeature>);
```

Comment:
Do we expect to have more information associated with each language feature that just present/not-present?
- If it's just present/not-present, we could use bitflags instead of a set.
    - Using bitflags would have the benefit of using much less space and avoiding `clone()` in a few places.
- If not, do we expect to be able to populate the additional information associated with each language feature from the CLI, manifest and other compiler APIs?
    - IMO, this option seems like a lot of complexity where a simple present/not-present would be enough.

Reply:
Using bitflags would also have the benefit of consistency since now runtime capabilities and language features would work the same way.